### ISSUE TITLE
Install the base-notebook and change Python version(not jupyter stack tag) .

### ISSUE THREAD

**Issue by amazongodman** [jupyter/docker-stacks#1208]
Hi, 
I read [this](https://github.com/jupyter/docker-stacks/issues/976).

and I want to change python_version by rewriting the dockerfile instead of looking for the tags in jupyter/notebook-stacks.

**What docker image you are using?**  

`jupyter/base-notebook`  

**What complete docker command do you run to launch the container (omitting sensitive values)?**  

git clone  jupyter/docker-stacks  
and edit Dockerfile 

>ARG PYTHON_VERSION=defolt
change to
>ARG PYTHON_VERSION=3.5

and
docker build -t py35basenote .

**What do you expect to happen?**

```
docker run -it py35basenote bash
python

>> pythonversion=3.5.x
```

**What actually happens?**

```
PackagesNotFoundError: The following packages are not available from current channels:

  - python=3.5.0

Current channels:

  - https://conda.anaconda.org/conda-forge/linux-64
  - https://conda.anaconda.org/conda-forge/noarch

```

Is this method wrong?
If you have ideas, I'd like to hear them.

**Comment by amazongodman** [comment]
when I set python=3.5  

```
SpecsConfigurationConflictError: Requested specs conflict with configured specs.
  requested specs:
    - conda-package-handling
    - python=3.5
  pinned specs:
    - conda==4.9.2
Use 'conda config --show-sources' to look for 'pinned_specs' and 'track_features'
configuration parameters.  Pinned specs may also be defined in the file
/opt/conda/conda-meta/pinned.
```

conda==4.9.2 is miss much?

**Comment by romainx** [comment]
Hello @amazongodman,

The version you are looking for `3.5.0` is not available in conda-forge.

```bash
$ conda search python=3.5 | head

# Loading channels: ...working... done
#  # Name                       Version           Build  Channel             
#  python                         3.5.1               0  conda-forge         
#  python                         3.5.1               1  conda-forge         
#  python                         3.5.2               0  conda-forge         
#  python                         3.5.2               1  conda-forge         
```

So the older version available for `3.5` is `3.5.1` and the latest is `3.5.5`.

If you want to use one of them you have not to edit the `Dockerfile` you can pass the pass the python version as an `ARG` at build time. But Python `3.5` seems to be in conflict with the version of conda `4.9.2`. I will not investigate further here to find the root cause because it's not the scope of this project. So my example below is using a more recent Python version, `3.6`. 

```bash
$ docker build --rm --force-rm \
    -t jupyter/base-notebook:python-3.6 ./base-notebook \
    --build-arg PYTHON_VERSION=3.6

$ docker run --rm -t jupyter/base-notebook:python-3.6 python --version

# Python 3.6.12
```

I think I've answered to your question. So I'm closing this issue. Feel free to post a new comment if something is missing.
Best

**Comment by amazongodman** [comment]
Hello @romainx 

I never know that build-arg existed !
Thanks for sharing(teaching) that new knowledge!

I see that 3.5 conflicts with conda 4.9.2.
I'm curious because of my intellectual curiosity, so I'll try to find out for myself.

3.6 doesn't even have a tag in the jupyter repository in docker hub, so this is very valuable advice!
